torch/_inductor/cudagraph_trees.py

```diff
@@ -582,14 +585,18 @@ def run(self, new_inputs):
         existing_path_data_ptrs = {
             t.data_ptr() for t in self.path_live_weakrefs() if t()
         }
-        non_cudagraph_inps = set()
-        for i in range(len(new_inputs)):
-            if (
-                isinstance(new_inputs[i], torch.Tensor)
-                and new_inputs[i].untyped_storage().data_ptr()
-                not in existing_path_data_ptrs
-            ):
-                non_cudagraph_inps.add(new_inputs[i].untyped_storage().data_ptr())
+
+        def get_non_cudagraph_inps():
+            non_cudagraph_inps = set()
+            for t in itertools.chain(new_inputs, self.wrapped_function.constants):
+                if (
+                    isinstance(t, torch.Tensor)
+                    and t.untyped_storage().data_ptr() not in existing_path_data_ptrs
+                ):
+                    non_cudagraph_inps.add(t.untyped_storage().data_ptr())
+            return non_cudagraph_inps
```

Comment:
poor man's scoping?

Reply:
previously, i had this inlined and at the end `del t` but `t` is not always set... better suggestion ?